### PR TITLE
Implementation of break lines, sub & superscripts as well as bold & italic styles

### PR DESCRIPTION
**Pull request by archmoj**
This PR enables the use of tags namely < b > < i > < sup > < sub> and < br > within webGL texts.
Please also refer to [Plotly PR 3207](https://github.com/plotly/plotly.js/pull/3207) for user issues, examples & more information.
@mikolalysenko 
@monfera

### CODE REVIEW

**Comment by mikolalysenko** [comment]
Could we get an option to make the styling stuff only run if you turn on a flag?  Not sure how much stuff this would break if we put it in by default all the time.

Maybe have an option for "style" just like the newlines code.

File: lib/vtext.js [hunk 1]
```diff
@@ -83,27 +149,172 @@ function transformPositions(positions, options, size) {
   })
 }
 
-function getPixels(canvas, context, str, size) {
-  var width = Math.ceil(context.measureText(str).width + 2*size)|0
-  if(width > 8192) {
-    throw new Error("vectorize-text: String too long (sorry, this will get fixed later)")
+function getPixels(canvas, context, rawString, fontSize, lineSpacing, styletags) {
+
+  console.log("rawString=", rawString);
```
Comment (etpinard) [inline lib/vtext.js]: :zap: please.

Comment (archmoj) [reply lib/vtext.js]: Thanks @etpinard for the note : )

**Comment by archmoj** [comment]
I still need to setup few other things here.
PR Status: `in progress`

File: lib/vtext.js [hunk 2]
```diff
@@ -190,8 +398,50 @@ function processPixels(pixels, options, size) {
...
-  
+
+  var size = 64
+  var lineSpacing = 1.25
+  var styletags = {
+    breaklines: false, // Note: could also be turned on as new default
+    bolds: false,      // Note: could also be turned on as new default
+    italics: false,    // Note: could also be turned on as new default
+    subscripts: false, // Note: could also be turned on as new default
+    superscripts: true // Note: superscripts used to be partly supported
```
Comment (archmoj) [inline lib/vtext.js]: We actually didn't support superscripts in this module. Anyway, what do you think @mikolalysenko @etpinard about the default behaviour of styling tags? Should we keep all of them off? Or if you prefer to enable any or the rest by default?

**Comment by archmoj** [comment]
@mikolalysenko 
Your suggestion of having style options was very good. 
FYI - I also disabled them by default and documented this in the readme.
Status of this PR (the `vectorize-text` module) is now `reviewable`. 
With regards.

**Comment by archmoj** [comment]
Thanks @mikolalysenko for your review & the merge.
Once I tried to bump & release a minor version; I noticed that I don't have github & possibly npm permissions for this package.
I don't know if @bpostlethwaite @etpinard could also help me with this?

**Comment by mikolalysenko** [comment]
Sorry, I forgot to publish to npm.  My internet died while I was trying to
publish and I got distracted.

**Comment by archmoj** [comment]
Thanks @mikolalysenko for the new release.

**Pull request merged**